### PR TITLE
update dependencies and node versions

### PR DESCRIPTION
**Pull request by ralphtheninja**
Note that `nsp` was not updated, since it will not work on node version below 6.

Supersedes https://github.com/prebuild/prebuild-install/pull/27

### CODE REVIEW

**Comment by ralphtheninja** [comment]
Ok. Seems `standard` found some deprecated methods and also complaints on `0.10` and `0.12` because of updated `simple-get`.

File: test/download-test.js [hunk 1]
```diff
@@ -12,52 +12,17 @@ var build = path.join(__dirname, 'build')
...
-
-  var mkdirCount = 0
-  var _mkdir = fs.mkdir.bind(fs)
-  fs.mkdir = function () {
-    var args = [].slice.call(arguments)
-    if (mkdirCount++ === 0) {
-      t.equal(args[0], util.prebuildCache(), 'fs.mkdir called for prebuildCache')
-    }
-    _mkdir.apply(fs, arguments)
-  }
```
Comment (ralphtheninja) [inline test/download-test.js]: I removed these tests. They merely checks that these calls were made, which basically tests internal functionality. We could remove more of these for `.create{Read,Write}Stream` as well, although some of them are useful.

Let me know what you guys think.

File: test/download-test.js [hunk 2]
```diff
@@ -12,52 +12,17 @@ var build = path.join(__dirname, 'build')
...
-
   var writeStreamCount = 0
-  var _createWriteStream = fs.createWriteStream.bind(fs)
```
Comment (ralphtheninja) [inline test/download-test.js]: No need to bind.

File: test/download-test.js [hunk 3]
```diff
@@ -89,54 +54,26 @@ test('downloading from GitHub, not cached', function (t) {
...
-      t.equal(path, npmCache, 'fs.exists called for npm cache')
-      _exists(path, cb)
-    } else {
-      t.equal(path, cachedPrebuild, 'fs.exists called for prebuild')
-      _exists(path, function (exists) {
-        t.equal(exists, true, 'prebuild should be cached')
-        cb(exists)
-      })
-    }
-  }
```
Comment (ralphtheninja) [inline test/download-test.js]: Same as above.

**Comment by ralphtheninja** [comment]
This should be released as a new major, since we removed support for `0.10` and `0.12`

**Comment by mathiask88** [comment]
https://github.com/prebuild/prebuild-install/pull/27 is still up to date, only one `let` too much :P

**Comment by ralphtheninja** [comment]
> #27 is still up to date, only one let too much :P

I know but you also rewrote using const and arrow functions, which @mafintosh didn't like.

**Comment by mathiask88** [comment]
Okay, worth a try :)

File: download.js [hunk 4]
```diff
@@ -26,23 +26,23 @@ function downloadPrebuild (opts, cb) {
   if (opts.nolocal) return download()
 
   log.info('looking for local prebuild @', localPrebuild)
-  fs.exists(localPrebuild, function (exists) {
```
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply download.js]: `.exists` is deprecated since node version 4

[the remark this answers is not in the document]
Comment (juliangruber) [reply download.js]: thanks! :)

**Comment by ralphtheninja** [comment]
@mathiask88 Maybe you want to make a PR for `Object.assign()` now that we no longer support 0.x? Getting rid of a dependency is always a good idea.

**Comment by mathiask88** [comment]
@ralphtheninja Sure, I will open a PR tomorrow morning

**Pull request merged**